packages/suite/src/reducers/wallet/coinjoinReducer.ts

```diff
@@ -556,3 +552,51 @@ export const selectIsCoinjoinBlockedByAmountsTooSmall = memoizeWithArgs(
             .every(utxo => new BigNumber(utxo.amount).lt(minAllowedInputWithFee));
     },
 );
+
+// see https://github.com/trezor/trezor-suite/issues/7388#issue-1532836062
+export const selectRoundsNeeded = memoizeWithArgs(
+    (state: CoinjoinRootState, accountKey: AccountKey) => {
+        const account = selectAccountByKey(state, accountKey);
+        const targetAnonymity = selectCurrentTargetAnonymity(state) || 0;
+
+        const anonymitySet = account?.addresses?.anonymitySet || {};
+        const utxos = account?.utxo || [];
+
+        if (!utxos) {
+            return 0;
+        }
+
+        const anonymityTimesAmountSum = BigNumber.sum(
+            ...utxos.map(utxo =>
+                new BigNumber(utxo.amount).times(
+                    Math.min(targetAnonymity, anonymitySet[utxo.address] || 1),
+                ),
+            ),
+        );
+
+        const amountsSum = BigNumber.sum(...utxos.map(utxo => utxo.amount));
+
+        const accountAnonCapped = amountsSum.isZero()
```

Comment:
should be and integer?

Reply:
I think so, so it can be subtracted from targetAnonymity. That is why I call toNumber() here.

Reply:
Afaik `toNumber()` [doesn't convert](https://mikemcl.github.io/bignumber.js/#toN) to an integer 🤔

Reply:
Now I get it, you mean integer instead of float? I've thought about it, but I don't see why.

Reply:
Because the anonymity level is always an integer I thought but now I see that it is not that significant at all :D There's `Math.ceil` afterwards.